app/controllers/books/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Books
+  class CommentsController < ApplicationController
+    def create
+      book = Book.find(params[:book_id])
+      comment = book.comments.new(comments_params)
+      comment.save!
+      redirect_to book_path(book)
+    end
+
+    def destroy
+      book = Book.find(params[:book_id])
+      comment = book.comments.where(user: current_user).find(params[:id])
```

Comment:
この書き方でも全然よいですが、僕だったらシンプルに

```suggestion
      comment = current_user.comments.find(params[:id])
```

って書きますね。
日報のidが指定される可能性もゼロではないですが、悪意を持ってそういう無意味なことをやる人はいないと思うので。